app/src/main/java/org/permanent/permanent/ui/myFiles/AddOptionsFragment.kt

```diff
@@ -0,0 +1,122 @@
+package org.permanent.permanent.ui.myFiles
+
+import android.app.Activity
+import android.content.ClipData
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.net.Uri
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.lifecycle.ViewModelProvider
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment
+import org.permanent.permanent.Constants
+import org.permanent.permanent.PermissionsHelper
+import org.permanent.permanent.R
+import org.permanent.permanent.databinding.FragmentAddOptionsBinding
+import org.permanent.permanent.viewmodels.AddOptionsViewModel
+
+class AddOptionsFragment: BottomSheetDialogFragment(), View.OnClickListener {
+
+    private lateinit var binding: FragmentAddOptionsBinding
+    private lateinit var viewModel: AddOptionsViewModel
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        binding = FragmentAddOptionsBinding.inflate(inflater, container, false)
+        binding.executePendingBindings()
+        binding.lifecycleOwner = this
+        viewModel = ViewModelProvider(this).get(AddOptionsViewModel::class.java)
+        binding.viewModel = viewModel
+        binding.btnUpload.setOnClickListener(this)
+
+        return binding.root
+    }
+
+    override fun onClick(view: View) {
```

Comment:
This can be done directly in xml

Reply:
could not move because requesting files reading permission needs fragment context, so as starting the file selection activity :(